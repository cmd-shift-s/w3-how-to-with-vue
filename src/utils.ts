import MarkdownIt from 'markdown-it'
import hljs from 'highlight.js'
import 'highlight.js/styles/vs.css'

export const md = new MarkdownIt({
  highlight (code, lang) {
    return hljs.highlight(lang, code).value
  }
})
