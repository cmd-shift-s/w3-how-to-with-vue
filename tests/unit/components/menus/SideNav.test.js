import { mount, renderer } from '@/tests/unit/utils'
import SideNav from '@/components/menus/SideNav.vue'

describe('SideNav.vue', () => {
  it('render default', () => {
    const wrap = mount(SideNav)

    expect(wrap.classes()).toContain('side-nav')
    expect(wrap.find('.close-btn').is('a')).toBe(true)

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('props#active', () => {
    const wrap = mount(SideNav)

    wrap.setProps('active', true)
    wrap.update()

    wrap.find('.close-btn').trigger('click')

    expect(wrap.emitted()['update:active'][0]).toEqual([false])
    expect(wrap.emitted().close).toBeTruthy()
  })

  it('props#is-animated', () => {
    const wrap = mount(SideNav, {
      propsData: {
        isAnimated: true
      }
    })

    expect(wrap.classes()).toContain('is-animated')

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('props#text-center', () => {
    const wrap = mount(SideNav, {
      propsData: {
        textCenter: true
      }
    })

    expect(wrap.classes()).toContain('has-text-centered')

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
