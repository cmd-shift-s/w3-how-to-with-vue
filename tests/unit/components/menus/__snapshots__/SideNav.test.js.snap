// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav.vue props#is-animated 1`] = `
<div class="side-nav is-animated">
    <div class="menus"> <a class="close-btn">×</a></div>
</div>
`;

exports[`SideNav.vue props#text-center 1`] = `
<div class="side-nav has-text-centered">
    <div class="menus"> <a class="close-btn">×</a></div>
</div>
`;

exports[`SideNav.vue render default 1`] = `
<div class="side-nav">
    <div class="menus"> <a class="close-btn">×</a></div>
</div>
`;
